lib/connectors/memory.js

```diff
@@ -679,6 +681,52 @@ Memory.prototype.updateAttributes = function updateAttributes(model, id, data, o
   }
 };
 
+Memory.prototype.replace = function replace(model, id, data, options, cb) {
+  var self = this;
+  if (!id) {
+    var err = new Error('You must provide an id when replacing!');
+    if (cb) {
+      return cb(err);
+    } else {
+      throw err;
+    }
+  }
+
+  // Do not modify the data object passed in arguments
+  // data = Object.create(data);
```

Comment:
```
var y = Object.create({x:1});
console.log(y); // {}
console.log(y.x); // 1 
```
Please note the {x: 1} will be set as the prototype of y. As a result, the x property is not owned by y but it's on the prototype chain.